graphql_compiler/tests/test_input_data.py

```diff
@@ -319,6 +319,85 @@ def between_filter_on_datetime():
         type_equivalence_hints=None)
 
 
+def between_lowering_on_simple_scalar():
+    # The "between" filter emits different output depending on what the compared types are.
+    # This test checks for correct code generation when the type is a simple scalar (a String).
+    graphql_input = '''{
+        Animal {
+            name @filter(op_name: "<=", value: ["$upper"])
+                 @filter(op_name: ">=", value: ["$lower"])
+                 @output(out_name: "name")
+        }
+    }'''
+    expected_output_metadata = {
+        'name': OutputMetadata(type=GraphQLString, optional=False),
+    }
+    expected_input_metadata = {
+        'lower': GraphQLString,
+        'upper': GraphQLString,
+    }
+
+    return CommonTestData(
+        graphql_input=graphql_input,
+        expected_output_metadata=expected_output_metadata,
+        expected_input_metadata=expected_input_metadata,
+        type_equivalence_hints=None)
+
+
+def between_lowering_with_extra_filters():
+    graphql_input = '''{
+        Animal {
+            name @filter(op_name: "<=", value: ["$upper"])
+                 @filter(op_name: "has_substring", value: ["$substring"])
+                 @filter(op_name: "in_collection", value: ["$fauna"])
+                 @filter(op_name: ">=", value: ["$lower"])
+                 @output(out_name: "name")
+        }
+    }'''
+    expected_output_metadata = {
+        'name': OutputMetadata(type=GraphQLString, optional=False),
+    }
+    expected_input_metadata = {
+        'lower': GraphQLString,
+        'upper': GraphQLString,
+        'substring': GraphQLString,
+        'fauna': GraphQLList(GraphQLString)
+    }
+
+    return CommonTestData(
+        graphql_input=graphql_input,
+        expected_output_metadata=expected_output_metadata,
+        expected_input_metadata=expected_input_metadata,
+        type_equivalence_hints=None)
+
+
+def no_between_lowering_on_simple_scalar():
+    # The "between" filter emits different output depending on what the compared types are.
+    # This test checks for correct code generation when the type is a simple scalar (a String).
```

Comment:
I know we discussed this in person, but can you add an explanation of why we decided not to lower to `BETWEEN` in this situation? Neither of us will remember the conversation or our reasoning in 6 months' time.